proto/warehouse/warehouse.proto

```diff
@@ -109,11 +113,69 @@ message RetryWHUploadsResponse {
 }
 
 message ValidateObjectStorageRequest {
-  string type=1;
-  google.protobuf.Struct config=2;
+  string type = 1;
+  google.protobuf.Struct config = 2;
 }
 
 message ValidateObjectStorageResponse{
-  bool isValid=2;
-  string error=3;
+  bool isValid = 2;
+  string error = 3;
+}
+
+message RetrieveFailedBatchesRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+}
+
+message RetrieveFailedBatchesResponse{
+  repeated FailedBatchInfo infos = 1;
+}
+
+message FailedBatchInfo{
+  string errorCategory = 1;
+  string sourceID = 2;
+  int64 count = 3;
+  google.protobuf.Timestamp lastHappened = 4;
+  string status = 5;
+}
+
+message RetrieveFailedBatchRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+  string errorCategory = 4;
+  string sourceID = 5;
+  string status = 6;
+}
+
+message RetrieveFailedBatchResponse{
+  string error = 1;
+  string errorCategory = 2;
+  string sourceID = 3;
```

Comment:
Do we need to return the `sourceID` here? We have it in the request so we can assume the client can retain it and not transfer it.

Reply:
Upon discussion with the pipelines team, we are sending an error in the batch request instead of exposing a separate API.